platforms/core-execution/persistent-cache/src/integTest/groovy/org/gradle/cache/internal/DefaultFileLockManagerContentionIntegrationTest.groovy

```diff
@@ -53,7 +54,7 @@ class DefaultFileLockManagerContentionIntegrationTest extends AbstractIntegratio
         executer.requireOwnGradleUserHomeDir().withDaemonBaseDir(file("daemonsRequestingLock")).requireDaemon()
         buildFile << """
             import org.gradle.cache.FileLockManager
-            import org.gradle.cache.internal.filelock.LockOptionsBuilder
+            import ${DefaultLockOptions.name}
```

Comment:
It was a little annoying having to find these references when doing renames.  This makes the class show up as an import and follows renames in the IDE.  You still have to fix references elsewhere, but the error you get is a bit better. There are more examples of this below in "worker not timeout".  Maybe all script imports should be done this way for consistency. 

There might be some better ways to deal with this.  I tried using the Groovy DSL test fixture that does syntax highlighting, but it doesn't seem to work with this complicated script.

Reply:
That makes sense.  I agree this error for a valid import paired with usage of a class name that doesn't align with it is probably easier to decipher than the class not found error if the (invalid) import and usage do align though.

Bigger picture, I suppose I'd ideally like to use a plain string to write the `buildFile` and then say to the test fixture - BTW, I'm importing these classes and give it some class tokens.  Then have the runner automatically catch class not found exceptions when compiling the scripts, check the given list of tokens, and provide an error like "did you rename the missing org.gradle.X class to org.gradle.something.Y?", finding the most similar name of the given candidates somehow.

Sounds just a little out of scope here 😛.

Reply:
I looked at this again because I thought this did work and I figured out what I was doing wrong.

1. If you use `buildFile << """...` then you don't get any syntax highlighting. You must use `buildFile """ ...`
2. If you have _any_ string interpolation, the syntax highlighting will give up.

Once I got syntax highlighting working, then IntelliJ correctly identified the imports.

It's not perfect though. It doesn't help you add new imports and it still doesn't follow renames of types. But you can click on bits of the code and go into those classes and the tests show up as usages.